internal/provisioner/plain/plain.go

```diff
@@ -24,15 +26,25 @@ const (
 	manifestsDir = "manifests"
 )
 
-func HandleBundle(_ context.Context, fsys fs.FS, _ *rukpakv1alpha1.Bundle) (fs.FS, error) {
-	if err := ValidateBundle(fsys); err != nil {
-		return nil, err
+func HandleBundleDeployment(_ context.Context, fsys fs.FS, bd *rukpakv1alpha2.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
+	isregistryV1, err := isRegistryV1Bundle(bd)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	processedFs := fsys
+	if isregistryV1 {
+		processedFs, err = convert.RegistryV1ToPlain(fsys)
+		if err != nil {
+			return nil, nil, fmt.Errorf("convert registry+v1 bundle to plain+v0 bundle: %v", err)
+		}
+	}
```

Comment:
Ah, I see. This answers my question about whether we need a separate handler. But this raises another question.

If we have the exact same logic between registry+v1 and plain provisioners, do we need both provisioners?